File: core/memory/semantic_history_engine.py
from __future__ import annotations

from typing import Any, Dict, List


def record_history(events: List[Dict[str, Any]], max_events: int = 500) -> Dict[str, Any]:
    bounded = events[-max_events:]
    return {"events": bounded, "count": len(bounded), "truncated": len(events) > max_events}

File: src/memory/semanticHistoryEngine.ts
/**
 * Converted from Python: core/memory/semantic_history_engine.py
 * @generated — WebWeaveX python→javascript library port
 */

import * as py from "../runtime/pyCompat.js";

export function recordHistory(events: any, max_events: any = 500): any {
  var bounded: any = py.slice(events, (-max_events), null);
  return {"events": bounded, "count": py.len(bounded), "truncated": (py.len(events) > max_events)};
}
